### PR TITLE
Feature/#519 frontend dependencies

### LINKED ISSUE #519: Copy frontend dependencies out of node_modules
Now that everything is installed into `/node_modules/`, we need to figure out a solid way of moving/copying frontend dependencies (for lack of a better term) out of `/node_modules/` so that they are available in the theme.

The main problem:
- We don't commit `/node_modules/`
- So, when we add something like Slick Carousel via `npm`, it gets installed in `/node_modules/` but will never be committed
- Thanks to @gregrickaby we have a bit of a solution in our `package.json` file:
  ```"copy:slick": "cp -R ./node_modules/slick-carousel/ assets/slick-carousel || true",```

This works, _sort of_. @bezierer  has run into an issue where a package he needed was being copied into a folder of itself. Instead of being copied into `/assets/package-name/` it was being copied into `/assets/package-name/package-name/`.

Currently, the `copy:slick` script runs on `npm run build`. A workaround here is to remove it from `npm run build` and let whoever is adding the package copy it once and then commit it. Then, it's there any time anyone else pulls down the repo.

The bummer part of this is:
- It's a workaround
- It involves extra steps
- It's clunky
- It's a little finicky and doesn't give us an easy way to copy an updated version to overwrite the existing version

I found a package called **frontend-dependencies** here: https://www.npmjs.com/package/frontend-dependencies

On the surface, it sounds great and exactly like what we need. I tried messing around with it a bit yesterday, though, and couldn't get it to do what I needed it to do. Might've just been the end of a long day and my brain was fried.

Would love some extra 👀 and :brain: on this to come up with a better solve for this issue!

### PR DESCRIPTION
**Pull request by coreymcollins**
Closes #519

### DESCRIPTION ###
Adds [frontend-dependencies](https://www.npmjs.com/package/frontend-dependencies) to manage our... _frontend dependencies_.

This gets rid of the manual copy/paste script we had running and smartly copies our frontend deps over into an `/assets/` directory (this name can be anything we want, if we want to change it).

I haven't been able to figure out if there's a way to install packages to the `frontendDependencies` object the same way you install them to `devDependencies`/`dependencies`. Maybe I'm just missing something, but I wanted to at least get a PR spun up that we can work from.

Right now, if you add your desired package to the object it'll be installed when you run `npm i` or `npm run build`.

This also works well with our `clean` script, so if we update the version number of a frontend dependency we'll just need to run `npm i` or `npm run build` to clear out the entirety of `/assets/` and install the new versions.

### SCREENSHOTS ###
package file

directory structure

### OTHER ###
- [x] Is this issue accessible? (Section 508/WCAG 2.0AA)
- [x] Does this issue pass all the linting? (PHPCS, ESLint, SassLint)
- [x] Does this pass CBT?

### STEPS TO VERIFY ###
1. Checkout the branch
2. Run `npm i`
3. Rejoice
4. Run `npm run build` to see the clean and install processes run
5. Test adding some other packages to the `frontendDependencies` object to see multiple packages copied over.

### DOCUMENTATION ###
Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

For sure. We'll need to detail how to install packages with this.

### CODE REVIEW

**Comment by gregrickaby** [comment]
@coreymcollins I played around and made some modifications. 

- Moved Slick to `/dist`. The static assets are already served from this directory, made sense to move Slick there too.
- Tweaked our scripts a bit. There's no need to lint on a build (we're linting on `watch`) and this slows down Buddy.
- Bumped dependencies

**Comment by coreymcollins** [comment]
Nice work! We just chatted about this, and I'll add some updates here for the following bits and pieces:
- Copy the whole of Slick's directory to encompass the license and other top-level files
- Remove the `composer` script from our package file
- Clean out the whole of `/dist/` when `clean` is run
- Run `frontend` on `npm run build` as well, otherwise we're only cleaning `/dist/` but never putting our frontend deps back when we run `npm run build`. Without this adjustment, the frontend deps only get added on `postinstall`
- Double check references/enqueues to Slick to make sure the directories are all updated

Thanks!

**Comment by coreymcollins** [comment]
@gregrickaby I think this is ready for another review with some updates made above.

I did add Slick back to the `dependencies` object in our package file. The reason for this is twofold, but I'm not married to it if you think we can do without it:
- This ensures it's going to be installed and that Frontend Dependencies will have it to copy from. I don't know that this would become an issue, but thought it might be a better safe than sorry situation?
- This removes the guesswork for engineers who aren't sure which version or version number syntax to use in the `frontendDependencies` object. If Slick is always installed in `dependencies`, or a new package is installed, the engineers will be able to copy the line from `dependencies` to `frontendDependencies` verbatim without having to make sure they're getting the right version – the right version would have already been installed as a dependency.

This also assumes there isn't a way to install a package directly to the `frontendDependencies` object, which I haven't been able to find yet.

I've tried different variations of flags like `npm install slick-carousel --save-frontend-dependencies` and the like and looked through their wiki/readme/repo but haven't found a way to add packages in such a way.

Lastly, if a package in our dependencies is out of date and we want to install a new version, running `npm install slick-carousel@latest` will update the version number in `dependencies` – it does not update the version number in `frontendDependencies`. I think this is another positive for keeping packages in `dependencies`, as we'll always be able to see if the version is ahead `frontendDependencies` at a glance so the version number can be updated.

Open to discussion on all of this, of course!

**Comment by gregrickaby** [comment]
@coreymcollins Sounds good and makes sense. I think this is 👍

**Review by gregrickaby — APPROVED** [review]

File: inc/scripts.php [hunk 1]
```diff
@@ -74,8 +74,8 @@ function _s_scripts() {
 
 	// Register styles & scripts.
 	wp_register_style( '_s-google-font', _s_font_url(), array(), null ); // @codingStandardsIgnoreLine - required to avoid Google caching issues.
-	wp_register_style( 'slick-carousel', get_template_directory_uri() . '/assets/slick-carousel/slick/slick.css', null, '1.8.1' );
-	wp_register_script( 'slick-carousel', get_template_directory_uri() . '/assets/slick-carousel/slick/slick' . $suffix . '.js', array( 'jquery' ), '1.8.1', true );
+	wp_register_style( 'slick-carousel', get_template_directory_uri() . '/dist/slick/slick.css', null, '1.8.1' );
```
Comment (vegasgeek) [inline inc/scripts.php:77]: @coreymcollins the path for lines 77 & 78 should be /dist/slick-carousel/slick/

**Pull request merged**